flexget/plugins/clients/deluge.py

```diff
@@ -690,9 +690,17 @@ def on_fail(result, task, entry):
         # dlist is a list of deferreds that must complete before we exit
         dlist = []
         # loop through entries to get a list of labels to add
-        labels = set([format_label(entry['label']) for entry in task.accepted if entry.get('label')])
-        if config.get('label'):
-            labels.add(format_label(config['label']))
+        labels = set()
+        for entry in task.accepted:
+            if entry.get('label', config.get('label')):
+                try:
+                    label = format_label(entry.render(entry.get('label', config.get('label'))))
+                    log.debug('Rendered label: %s', label)
+                except RenderError as e:
+                    log.error('Error rendering label `%s`: %s', label, e)
```

Comment:
Yes but if the render fails, `except` will cause it to continue to the next entry before it gets to using `label`.